ark-cli/src/commands/list.rs

```diff
@@ -0,0 +1,341 @@
+
+                let scores = if self.scores {
+                    Some(
+                        read_storage_value(
+                            &root,
+                            "scores",
+                            &resource.id.to_string(),
+                            &None,
+                        )
+                        .map_or(0, |s| s.parse::<u32>().unwrap_or(0)),
+                    )
+                } else {
+                    None
+                };
+
+                let datetime = if self.modified {
+                    let format = "%b %e %H:%M %Y";
+                    Some(
+                        DateTime::<Utc>::from(resource.modified)
+                            .format(format)
+                            .to_string(),
+                    )
+                } else {
+                    None
+                };
+
+                let (path, resource, content) = match entry_output {
+                    EntryOutput::Both => (
+                        Some(path.to_owned().into_path_buf()),
+                        Some(resource.clone().id),
+                        None,
+                    ),
+                    EntryOutput::Path => {
+                        (Some(path.to_owned().into_path_buf()), None, None)
+                    }
+                    EntryOutput::Id => (None, Some(resource.clone().id), None),
+                    EntryOutput::Link => match File::open(path) {
+                        Ok(mut file) => {
+                            let mut contents = String::new();
+                            match file.read_to_string(&mut contents) {
+                                Ok(_) => (None, None, Some(contents)),
+                                Err(_) => return None,
+                            }
+                        }
+                        Err(_) => return None,
+                    },
+                };
+
+                Some(StorageEntry {
+                    path,
+                    resource,
+                    content,
+                    tags,
+                    scores,
+                    datetime,
+                })
+            })
+            .collect::<Vec<_>>();
+
+        match self.sort {
+            Some(Sort::Asc) => {
+                storage_entries.sort_by(|a, b| a.datetime.cmp(&b.datetime))
+            }
+
+            Some(Sort::Desc) => {
+                storage_entries.sort_by(|a, b| b.datetime.cmp(&a.datetime))
+            }
+            None => (),
+        };
+
+        if let Some(filter) = &self.filter {
+            storage_entries.retain(|entry| {
+                entry
+                    .tags
+                    .as_ref()
+                    .map(|tags| tags.contains(filter))
+                    .unwrap_or(false)
+            });
+        }
+
+        let no_tags = "NO_TAGS";
+        let no_scores = "NO_SCORE";
+
+        // FIXME: We should use a table library to print this.
+        //        probably `comfy-table`
```

Comment:
We have such issue in this repo: https://github.com/ARK-Builders/ark-rust/issues/51